event-processing/event-processing-template.yml

```diff
@@ -2953,3 +2981,279 @@ Resources:
+
+  PublishToAccountsSNSSubscription:
+    Type: AWS::SNS::Subscription
+    Properties:
+      TopicArn: !Ref epSNSTopic
+      Endpoint: !GetAtt  PublishToAccountsRedactedFunction.Arn #Lambda
+      Protocol: lambda
+      FilterPolicy:
+        eventName:
+          - AUTH_AUTH_CODE_ISSUED
+
+  PublishToAccountsPolicy:
+    Type: AWS::IAM::Policy
+    Properties:
+      Roles:
+        - !Ref PublishToAccountsAccessRole
+      PolicyName: publish_to_accounts_policy
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action:
+              - 'sqs:SendMessage'
+            Resource:
+              - !Sub "${PublishToAccountsSQSQueue.Arn}"
+          - Effect: Allow
+            Action:
+              - 'sqs:DeleteMessage'
+              - 'sqs:ReceiveMessage'
+              - 'sqs:GetQueueAttributes'
+              - 'sqs:SendMessage'
+            Resource:
+              - !Sub "${PublishToAccountsSQSDLQueue.Arn}"
+          - Effect: Allow
+            Action:
+              - 'kms:Decrypt'
+              - 'kms:Encrypt'
+              - 'kms:GenerateDataKey*'
+              - 'kms:DescribeKey'
+            Resource: !GetAtt SQSKMSKeyPublishAccount.Arn
+
+  PublishToAccountsAccessRole:
+    Type: AWS::IAM::Role
+    Properties:
+      PermissionsBoundary: !If
+        - UsePermissionsBoundary
+        - !Ref PermissionsBoundary
+        - !Ref AWS::NoValue
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - sqs.amazonaws.com
+                - lambda.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      ManagedPolicyArns:
+        - "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+
+  PublishToAccountsRedactedLogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      KmsKeyId: !GetAtt LogsKmsKey.Arn
+      LogGroupName: "/aws/lambda/PublishToAccountsRedactedFunction"
+      RetentionInDays: 7
+
+  PublishToAccountsRedactedFunctionPermission:
+    Type: AWS::Lambda::Permission
+    Properties:
+      Action: lambda:InvokeFunction
+      FunctionName: !Ref PublishToAccountsRedactedFunction
+      Principal: sns.amazonaws.com
+      SourceAccount: !Sub "${AWS::AccountId}"
+      SourceArn: !Ref epSNSTopic
+
+  PublishToAccountsRedactedFunction:
+    DependsOn:
+      - PublishToAccountsRedactedLogGroup
+      - PublishToAccountsAccessRole
+      - PublishToAccountsSQSQueue
+    Type: AWS::Serverless::Function
+    Properties:
+      FunctionName: "PublishToAccountsRedactedFunction"
+      CodeUri: ../event-processing/lambda
+      PackageType: Zip
+      Handler: redacted-app.handler
+      Runtime: nodejs14.x
+      Timeout: 30
+      Role: !GetAtt  PublishToAccountsAccessRole.Arn
+      Environment:
+        Variables:
+          sqsArn: !GetAtt PublishToAccountsSQSQueue.Arn
+          defaultComponentId: "ACCOUNTS"
```

Comment:
the environment variable should really be called `PUBLISHTOACCOUNTSQUEUEARN`, more descriptive that way

Reply:
> the environment variable should really be called `PUBLISHTOACCOUNTSQUEUEARN`, more descriptive that way

DONE